app/src/main/java/com/p2p/wallet/auth/ui/username/ReservingUsernamePresenter.kt

```diff
@@ -13,21 +17,47 @@ class ReservingUsernamePresenter(
     ReservingUsernameContract.Presenter {
 
     private var checkUsernameJob: Job? = null
+    private var nextCheckAvailable: Boolean = true
 
     override fun checkUsername(username: String) {
-        checkUsernameJob?.cancel()
-        checkUsernameJob = launch {
+        if (nextCheckAvailable) {
```

Comment:
Этот флаг нам не нужен. 
```suggestion
            checkUsernameJob?.cancel()
            checkUsernameJob = launch {
                try {
                    delay(300)
                    interactor.checkUsername(username)
                    view?.showUnavailableName(username)
                } catch (e: CancellationException) {
                    Timber.w("Cancelled checkUsername request")
                } catch (e: Throwable) {
                    view?.showAvailableName(username)
                }
            }
```

Мы понимаем, что пользователю не нужно проверять username, пока он не перестал печатать имя. Если он дописал username и ждет, запрос должен пойти.